### PR TITLE
Remove URL and Hash from proposals, Add title, increase character limit

### LINKED ISSUE #1184: Update field constraints in governance proposals
In order to update the field constraints in governance proposals we need to agree on what the changes should be:

- https://github.com/vegaprotocol/vega/issues/5752

We need to understand:

- [x] limitations on storage etc.. on the chain (@ze97286 / @jeremyletang)
- [x] what the FE requirements look like (@campbellssource)
- [ ] What the community / builders club would deem as a good UX experience (@cdm / @daunatv)

### PR DESCRIPTION
**Pull request by campbellssource**
closes: https://github.com/vegaprotocol/specs/issues/1184

- [x] make the changes that were agreed
- [x] decide on optionality of fields on different proposals
- [x] add ACs

### CODE REVIEW

File: protocol/0028-GOVE-governance.md [hunk 1]
```diff
@@ -169,16 +169,13 @@ We introduce 2 new commands which require consensus (needs to go through the cha
...
 
 Every proposal transaction contains the following common fields:
-- a link to a text file in markdown format
-- a cryptographically secure hash (SHA3-512) of the text so that viewers can check that the text hasn't been changed since the proposal was submitted and
-- a description field to show a short title / something in case the link goes offline.
-
-The protocol (Vega core) is not expected to verify that the hash corresponds to the contents of the linked file. It is expected that any client tool that allows voting will do this at client level.
+- a title field to briefly describe the proposal that can be used when listing proposals
+- a description field to contain the additional details behind the proposal as well as some rationale
+If more details is required about the proposal, a proposer can reference immutable external resources in the proposal description. e.g. [IPFS](https://en.wikipedia.org/wiki/InterPlanetary_File_System) links.
```
Comment (ValentinTrinque) [inline protocol/0028-GOVE-governance.md:174]: Should we verify the content of the description for link pointing *mutable* resources ?

Comment (jeremyletang) [reply protocol/0028-GOVE-governance.md:174]: we verify nothing in the contents.

File: protocol/0028-GOVE-governance.md [hunk 2]
```diff
@@ -414,7 +405,7 @@ APIs should also exist for clients to:
...
 - [ ] If a new proposal is sucessfully submitted to the network (passing initial validation) the required participation rate and majority for success are defined and copied to the proposal and can be queried via APIs separately from the general network parameters. (<a name="0028-GOVE-036" href="#0028-GOVE-036">0028-GOVE-036</a>)
 - [ ] If a new proposal "P" is sucessfully submitted to the network (passing initial validation) the required participation rate and majority for success are defined and copied to the proposal. If an independent network parameter change proposal is enacted changing either required participation of majority then proposal "P" uses its own values for participation and majority; not the newly enacted ones.  (<a name="0028-GOVE-037" href="#0028-GOVE-037">0028-GOVE-037</a>)
-
+- [ ] All proposals with a title field that is empty, or not between 0 and 100 characters, will be rejected (<a name="0028-GOVE-039" href="#0028-GOVE-039">0028-GOVE-039</a>)
```
Comment (barnabee) [inline protocol/0028-GOVE-governance.md]: 0 chars is as useful as empty, so if we require non-empty it should be more…
Should the min. length be 1 or something more credible?

File: protocol/0028-GOVE-governance.md [hunk 3]
```diff
@@ -453,11 +444,9 @@ Below `*` stands for any of `asset, market, updateMarket, updateNetParam, freeFo
 
 
 ### Freeform governance proposals
-- [ ] A freeform governance proposal with a description field that is empty, or not between 0 and 255 characters, will be rejected (<a name="0028-GOVE-019" href="#0028-GOVE-019">0028-GOVE-019</a>)
-- [ ] A freeform governance must contain a hash field and it must not be null, but no other check is done to verify it (<a name="0028-GOVE-020" href="#0028-GOVE-020">0028-GOVE-020</a>)
-- [ ] A freeform governance must contain a link field and it must not be null, but no other check is done to verify it (<a name="0028-GOVE-021" href="#0028-GOVE-021">0028-GOVE-021</a>)
+- [ ] A freeform governance proposal with a description field that is empty, or not between 0 and 10,000 characters, will be rejected (<a name="0028-GOVE-019" href="#0028-GOVE-019">0028-GOVE-019</a>)
```
Comment (barnabee) [inline protocol/0028-GOVE-governance.md]: As above, should the min. length be 1, or more if it is populated?

File: protocol/0028-GOVE-governance.md [hunk 4]
```diff
@@ -453,11 +444,9 @@ Below `*` stands for any of `asset, market, updateMarket, updateNetParam, freeFo
 
 
 ### Freeform governance proposals
-- [ ] A freeform governance proposal with a description field that is empty, or not between 0 and 255 characters, will be rejected (<a name="0028-GOVE-019" href="#0028-GOVE-019">0028-GOVE-019</a>)
-- [ ] A freeform governance must contain a hash field and it must not be null, but no other check is done to verify it (<a name="0028-GOVE-020" href="#0028-GOVE-020">0028-GOVE-020</a>)
```
Comment (jeremyletang) [inline protocol/0028-GOVE-governance.md:457]: Could we have this AC replaced with one ensuring that the Title is specified and is < 100 chars?

File: protocol/0028-GOVE-governance.md (hunk 2 above)
Comment (edd) [inline protocol/0028-GOVE-governance.md]:
```suggestion
- [ ] All proposals with a title field that is empty, or not between 1 and 100 characters, will be rejected (<a name="0028-GOVE-039" href="#0028-GOVE-039">0028-GOVE-039</a>)
```

File: protocol/0028-GOVE-governance.md (hunk 3 above)
Comment (edd) [inline protocol/0028-GOVE-governance.md]:
```suggestion
- [ ] A freeform governance proposal with a description field that is empty, or not between 1 and 10,000 characters, will be rejected (<a name="0028-GOVE-019" href="#0028-GOVE-019">0028-GOVE-019</a>)
```

File: protocol/0028-GOVE-governance.md (hunk 4 above)
Comment (campbellssource) [reply protocol/0028-GOVE-governance.md:457]: done in https://github.com/vegaprotocol/specs/pull/1186/commits/0a9f0f2a1ce1de3301dafba2bd1496fa38166de2

File: protocol/0028-GOVE-governance.md (hunk 3 above)
Comment (campbellssource) [reply protocol/0028-GOVE-governance.md]: done here: https://github.com/vegaprotocol/specs/pull/1186/commits/0a9f0f2a1ce1de3301dafba2bd1496fa38166de2

File: protocol/0028-GOVE-governance.md (hunk 2 above)
Comment (campbellssource) [reply protocol/0028-GOVE-governance.md]: done here: https://github.com/vegaprotocol/specs/pull/1186/commits/0a9f0f2a1ce1de3301dafba2bd1496fa38166de2

**Review by barnabee — APPROVED** [review]

**Review by edd — APPROVED** [review]

**Pull request merged**